native_mate/promise.h

```diff
@@ -38,7 +40,7 @@ class Promise {
   v8::Isolate* isolate_;
 
  private:
-  v8::Local<v8::Promise::Resolver> resolver_;
+  scoped_refptr<RefCountedPersistent<v8::Promise::Resolver>> handle_;
```

Comment:
We should use `v8::Global` here. `RefCountedPersistent` is only for the cases when we want to copy persistent handles, the `Promise` just be a movable type.